src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -1859,7 +1861,10 @@ private static void parseJailCommand(CommandSender sender, Town town, String[] s
 						throw new TownyException(Translatable.of("msg_resident_has_not_played_long_enough_to_be_jailed", jailedResident.getName(), TimeMgmt.getFormattedTimeValue(time)));
 					}
 				}
-
+				// If using initialJailFee option check they can actually afford to jail someone.
+				if (initialJailFee >= 0 && !town.getAccount().canPayFromHoldings(initialJailFee))

```

Comment:
Before touching anything to do with the economy you must first confirm that TownyEconomyHandler.isActive().

Reply:
Method now confirms TownyEconomyHandler.isActive() is true